content/code_comments/code-comments/_index.md

```diff
@@ -0,0 +1,75 @@
+---
+content_type: project
+flavours:
+  - any_language
+prerequisites:
+  hard:
+    - /code_comments/introduction-to-code-commenting
+ready: true
+submission_type: link
+tags:
+  - comments
+title: Code comment evaluation
+---
+
+## Introduction
+
+Understanding the difference between good comments and bad comments is essential for effective code communication and collaboration. Good comments provide purposeful explanations, relevant information, and clarity, while avoiding redundancy and outdated content. By following these principles, developers can create comments that enhance code comprehension, facilitate maintenance, and promote a positive coding environment.
+
+## Comment Evaluation
+
+1.Here are some code examples, please rewrite the code in a clearer way. Please make sure you only make use of comments that are actually useful. You are welcome to change function and variable names.
+
+Python:
+
+```python
+def area_of_rectangle(length, width):
```

Comment:
This code looks very clear to me... I'm not sure what the learner is meant to do here.

Maybe a better thing would be to include some code examples with really excessive commenting and then ask the learner to clean it up.

eg:

```
a = 1 // assign the value 1 to the variable named a
c = a + b  // add a to b 
```

eg2:

```
# get the area of the triangle. a,b and c are the lehngths of the sides
def area_of_triangle(a,b,c):
     s = ....  # semiperimeter
     return  ....
```

Reply:
The examples above are things I have seen learners do a lot.  

What other problems do learners often have? Let's focus in on those problems. 

I am sure you have given people feedback about bad commenting before? How can you give the learners those lessons in this project?